File: dfs_bfs/boj/p1697.py
from sys import stdin
from collections import deque

n, k = map(int, stdin.readline().split())

MAX = 100000
time = [0] * (MAX+1)

que = deque([n])

while que:
    current = que.popleft()

    if current == k:
        print(time[current])
        break

    for i in (current-1, current+1, current*2):
        if 0 <= i <= MAX and not time[i]:
            time[i] = time[current] + 1
            que.append(i)

File: dfs_bfs/boj/P1697.java
package dfs_bfs.boj;
import java.util.*;

/**
 * P1697
 */
public class P1697 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int n = sc.nextInt();
        int k = sc.nextInt();

        int MAX = 100000;
        int[] time = new int[MAX+1];
        
        Queue<Integer> que = new LinkedList<>();
        que.add(n);
        
        int current, next;
        while (!que.isEmpty()) {
            current = que.poll();

            if (current == k) {
                System.out.println(time[current]);
                break;
            }

            for (int i = 0; i < 3; i++) {
                if (i == 0) {
                    next = current - 1;
                } else if (i == 1) {
                    next = current + 1;
                } else {
                    next = current * 2;
                }

                if (0 <= next && next <= MAX && time[next] == 0) {
                    time[next] = time[current] + 1;
                    que.add(next);
                }
            }
        }

        sc.close();
    }
}